File: Core/MEDIUM/Problem-1310/Python/XOR_Queries_of_a_Subarray.py
from typing import *
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        prefixXOR = [0] * (len(arr) + 1)
        for i in range(1, len(arr) + 1):
            prefixXOR[i] = prefixXOR[i - 1] ^ arr[i - 1]
        res = []

        for l, r in queries:
            res.append(prefixXOR[r + 1] ^ prefixXOR[l])
        
        return res

File: Core/MEDIUM/Problem-1310/Java/XOR_Queries_of_a_Subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int[] prefixXOR = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            prefixXOR[i] = prefixXOR[i - 1] ^ arr[i - 1];
        }

        int[] res = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int l = queries[i][0], r = queries[i][1];
            res[i] = prefixXOR[r + 1] ^ prefixXOR[l];
        }
        return res;
    }
}
